### PR TITLE
Add if_found kwarg to subdir().

### PR DESCRIPTION
**Pull request by jpakkane**
Extracted from the dolphin branch since this is useful on its own.

### CODE REVIEW

File: docs/markdown/Reference-manual.md [hunk 1]
```diff
@@ -1032,6 +1032,12 @@ current build file and in all subsequent build files executed with
 Note that this means that each `meson.build` file in a source tree can
 and must only be executed once.
 
+This function has one keyword argument.
+
+ - `if_found` takes one or several dependency objects and will only
+   recurse in the subdir if they all return `true` when queried with
+   `.is_found()`
```
Comment (nirbheek) [inline docs/markdown/Reference-manual.md]: This should be `found()`

File: mesonbuild/interpreter.py [hunk 2]
```diff
@@ -2486,6 +2487,11 @@ def func_subdir(self, node, args, kwargs):
             raise InvalidArguments('Must not go into subprojects dir with subdir(), use subproject() instead.')
         if self.subdir == '' and args[0].startswith('meson-'):
             raise InvalidArguments('The "meson-" prefix is reserved and cannot be used for top-level subdir().')
+        for i in listify(kwargs.get('if_found', [])):
```
Comment (nirbheek) [inline mesonbuild/interpreter.py]: `extract_as_list()` instead of `listify()`?

**Comment by amitdo** [comment]
It's probably too late, but what about using an actual if condition?

```
dep_a = dependency(...)
dep_b = dependency(...)
deps = dependencies([[dep_a, dep_b])
if deps.found()
   subdir('foo')
....
```

**Comment by amitdo** [comment]
@nirbheek, @jpakkane 
 Any comment?

**Comment by jpakkane** [comment]
There's a separate issue on doing `found` on all entries in an array somewhere, I think.

**Pull request merged**